clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -100,11 +100,10 @@ public RuntimeException exception() {
      * @param value corresponding value (or null if there is none)
      */
     public void complete(T value) {
-        if (isDone)
+        if (!isDone.compareAndSet(false, true))
```

Comment:
@norwood Great point. Maybe we need to collapse these three fields to a single `AtomicReference` and use `instanceof` to check whether it's an exception or not.